src/main/scala/mesosphere/marathon/core/launcher/impl/TaskOpFactoryImpl.scala

```diff
@@ -113,23 +109,21 @@ class TaskOpFactoryImpl @Inject() (
     // create a TaskBuilder that used the id of the existing task as id for the created TaskInfo
     new TaskBuilder(app, (_) => task.taskId, config).build(offer, resourceMatch, volumeMatch) map {
       case (taskInfo, ports) =>
-        val newTask = task.copy(
-          agentInfo = Task.AgentInfo(
-            host = offer.getHostname,
-            agentId = Some(offer.getSlaveId.getValue),
-            attributes = offer.getAttributesList.asScala
+        val launch = TaskStateOp.Launch(
+          appVersion = app.version,
+          status = Task.Status(
+            stagedAt = clock.now()
           ),
-          launched = Some(
-            Task.Launched(
-              appVersion = app.version,
-              status = Task.Status(
-                stagedAt = clock.now()
-              ),
-              networking = Task.HostPorts(ports)
-            )
-          )
-        )
-        taskOperationFactory.launch(taskInfo, newTask)
+          networking = Task.HostPorts(ports))
+
+        // FIXME (3221): something like reserved.launch(...): LaunchedOnReservation so we don't need to match?
+        task.update(launch) match {
```

Comment:
Yes, the problem here is that I have a `Task` and not a reserved, so it comes back to either matching or fixing the call hierarchy. I will tackle this during task update serialization. (all `FIXME(3221)` should have vanished when finishing 3221 …)